M2/autogen.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+echo "-- Generating configure script"
+autoreconf --verbose --force --install
+
+if test ! -f Macaulay2/editors/emacs/M2.el
+then
+    if test -d ../.git
```

Comment:
If you use git's worktree feature, `.git` can be a file with the address of the real `.git` directory in it, hence why I've been saying the check should be directory _or_ file.